test/unit_tests/CMakeLists.txt

```diff
@@ -14,16 +14,20 @@ get_filename_component(
   ${CMAKE_CURRENT_LIST_DIR}/../..
   REALPATH
 )
-get_filename_component(
-  TEST_DIR
-  ${DEVICE_OS_DIR}/test
-  REALPATH
-)
-get_filename_component(
-  THIRD_PARTY_DIR
-  ${DEVICE_OS_DIR}/third_party
-  REALPATH
+set(TEST_DIR ${DEVICE_OS_DIR}/test)
+set(THIRD_PARTY_DIR ${DEVICE_OS_DIR}/third_party)
+
+# Add Boost from git submodule
+set(Boost_USE_MULTITHREADED OFF)
+find_package(Boost
+  1.59.0 EXACT
+  REQUIRED
+  COMPONENTS program_options random regex system thread
```

Comment:
`random` is in this list because of `<particle-iot/device-os>/ci/build_boost.sh`, but based on the list in your first PR, we may not need to check for it.